### ISSUE TITLE
Minor issues

### ISSUE THREAD

**Issue by elvquant** [swesterfeld/audiowmark#1]
rawconverter.cc should "#include < array >"

testrandom.cc: two warnings: format specifies type 'unsigned long' but the argument has type 'uint64_t' (aka 'unsigned long long') [-Wformat]

audiowmark: libgcrypt version mismatch (GCRYPT_VERSION in random.cc?)
/opt/local/lib/libgcrypt.20.dylib (compatibility version 23.0.0, current version 23.1.0)

**Comment by swesterfeld** [comment]
From your included error message, I guessed that you are on macOS. I was able to reproduce the first two problems (array include, printf warning) and fixed it in

https://github.com/swesterfeld/audiowmark/commit/13d514203fe3205190415dfa99f54ab74915123c

I was not able to reproduce the libgcrypt issue. I believe that GCRYPT_VERSION is properly used in random.cc; mainly this means: require the gcrypt version we were compiled against. So my guess is that you are building against gcrypt version X and linking against gcrypt version Y. However, while fixing the problems, I also added the CFLAGS for libgcrypt. This may or may not fix your problem.
